extensions/pkg/controller/worker/actuator.go

```diff
@@ -13,17 +13,61 @@ import (
 type Actuator interface {
-	// Reconcile reconciles the Worker.
+	// Reconcile reconciles the [extensionsv1alpha1.Worker] resource.
+	//
+	// Implementations should ensure that any resources
+	// (e.g. ManagedResources, Secrets, etc.) are created or updated in
+	// order to reach their desired state.
 	Reconcile(context.Context, logr.Logger, *extensionsv1alpha1.Worker, *extensionscontroller.Cluster) error
-	// Delete deletes the Worker.
+
+	// Delete is invoked when the [extensionsv1alpha1.Worker] resource is
+	// deleted.
+	//
+	// Implementations should take care of cleaning up any resources
+	// (e.g. ManagedResources, Secrets, etc.), which were created by the
+	// Extension.
+	//
+	// Implementations must wait until all resources managed by the
+	// extension have been gracefully cleaned up.
 	Delete(context.Context, logr.Logger, *extensionsv1alpha1.Worker, *extensionscontroller.Cluster) error
-	// ForceDelete forcefully deletes the Worker.
+
+	// ForceDelete is invoked when the shoot cluster associated with the
+	// [extensionsv1alpha1.Worker] resource is being deleted in a forceful
+	// manner.
+	//
+	// Implementations should take care of unblocking the deletion flow by
+	// attempting to cleanup any resources created by the extension, remove
+	// any finalizers created for custom resources, etc., and also skip
+	// waiting for external resources, if they cannot be deleted gracefully.
+	//
+	// Even if some resources managed by the extension implementation cannot
+	// be deleted gracefully, this method should succeed, even at the cost
+	// of leaving some leftover resources behind.
 	ForceDelete(context.Context, logr.Logger, *extensionsv1alpha1.Worker, *extensionscontroller.Cluster) error
-	// Restore reads from the worker.status.state field and deploys the machines and machineSet
+
+	// Restore restores the [extensionsv1alpha1.Worker] from a previously
+	// saved.
+	//
+	// This method is invoked when the shoot cluster associated with the
+	// [extensionsv1alpha1.Worker] resource is being restored on the target
+	// seed cluster.
+	//
+	// Implementations may use the persisted data in the .status.state field
+	// for restoring the state, when the shoot is being migrated to a
+	// different seed cluster.
 	Restore(context.Context, logr.Logger, *extensionsv1alpha1.Worker, *extensionscontroller.Cluster) error
-	// Migrate deletes the MCM, machineDeployments, machineClasses, machineClassSecrets,
-	// machineSets and the machines. The underlying VMs representing the Shoot nodes are not deleted
+
+	// Migrate prepares the [extensionsv1alpha1.Worker] resource for
+	// migration.
+	//
+	// This method is invoked when the shoot cluster associated with the
+	// [extensionsv1alpha1.Extension] resource is being migrated to another
+	// seed cluster.
+	//
+	// Implementations should take care of storing any required state in the
+	// .status.state field, so that it can later be restored from this
+	// state.
```

Comment:
We should refer to the Gardenlet here which manages MCM and takes care about persisting the machine states when the shoot is being migrated (main job). Usually, there is nothing left to be implemented by the actuator.